cql3/restrictions/statement_restrictions.cc

```diff
@@ -1253,6 +1357,22 @@ bool token_known(const statement_restrictions& r) {
 bool statement_restrictions::need_filtering() const {
     using namespace expr;
 
+    if (has_token(_partition_key_restrictions->expression)) {
+        // If there is a token(p1, p2) restriction, no p1, p2 restrictions are allowed in the query.
+        // All other restrictions must be on clustering or regular columns.
```

Comment:
This limiation makes sense, but I never thought about it before. Do we have a test verifying it?

Reply:
Yes:
https://github.com/scylladb/scylla/blob/7261c2c93e633a5df2772cc1313da59712ba74d4/test/boost/restrictions_test.cc#L900-L905

Reply:
As for limitations on `token()` + index + some other, the [tests added in this pr](https://github.com/cvybhu/scylla/blob/6482528bd1021e5bde60acceef9959d072c6195e/test/boost/secondary_index_test.cc#L1725) should test this indirectly. If `need_filtering` didn't return `true` somewhere when it is needed then the result would be wrong.
I could maybe add separate tests to explicitly check when adding `ALLOW FILTERING` is needed when mixing `token()`, indexes and other restrictions in `test_allow_filtering.py`.